server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -1742,6 +1747,8 @@ private static IndexMetadata.Builder restoreOverClosedIndex(IndexMetadata snapsh
             .settingsVersion(Math.max(snapshotIndexMetadata.getSettingsVersion(), 1 + currentIndexMetadata.getSettingsVersion()))
             .aliasesVersion(Math.max(snapshotIndexMetadata.getAliasesVersion(), 1 + currentIndexMetadata.getAliasesVersion()))
             .timestampRange(IndexLongFieldRange.NO_SHARDS)
+            // TODO: we don't have access to cluster state here, so we can't check min transport version - will that be a problem?
```

Comment:
I think it will, since there are assertions validating that we are not sending this state to an older node. Seems we have the cluster state one stack frame out, so seems easily solvable?

Reply:
Good idea. Fixed in next push.